account_invoice_production_lot/models/account_invoice.py

```diff
@@ -31,10 +31,21 @@ class AccountInvoiceLine(models.Model):
         compute='_compute_line_lots',
     )
 
+    @api.multi
+    def _get_returned_moves(self):
+        self.ensure_one()
+        procurement_ids = self.move_line_ids.mapped('procurement_id').ids
+        moves = self.env['stock.move'].search(
+            [('procurement_id', 'in', procurement_ids)]).filtered(
+            lambda m: m.state == ('done'))
```

Comment:
Why don't you insert this in the domain?
One more question: here you are getting all the done moves linked to the procurement, how can you be sure that they are moves for returned products?

Reply:
> Why don't you insert this in the domain?

Done in https://github.com/OCA/account-invoice-reporting/pull/93/commits/d8e31422ebd29eb9632a5c957c93ec3020eecefa

> how can you be sure that they are moves for returned products?

Checking if `origin_returned_move_id` is filled (see https://github.com/odoo/odoo/blob/12.0/addons/stock/wizard/stock_picking_return.py#L87).
And by mapping as follows:

`moves.mapped('origin_returned_move_id')`

I get all the moves involved in the return.